pkg/resources/cruisecontrol/configmap.go

```diff
@@ -92,3 +94,86 @@ func generateBootstrapServer(headlessEnabled bool, clusterName string) string {
+	storageConfigNWINDefaultValue  = "10000"
+	storageConfigNWOUTDefaultValue = "10000"
+)
+
+type CruiseControlCapacityConfig struct {
+	BrokerCapacities []BrokerCapacity `json:"brokerCapacities"`
+}
+type BrokerCapacity struct {
+	BrokerID string   `json:"brokerId"`
+	Capacity Capacity `json:"capacity"`
+	Doc      string   `json:"doc"`
+}
+type Capacity struct {
+	DISK  map[string]string `json:"DISK"`
+	CPU   string            `json:"CPU"`
+	NWIN  string            `json:"NW_IN"`
+	NWOUT string            `json:"NW_OUT"`
+}
+
+// generateCapacityConfig generates a CC capacity config with default values or returns the manually overridden value if it exists
+func GenerateCapacityConfig(kafkaCluster *v1beta1.KafkaCluster, log logr.Logger) string {
+	// If there is already a config added manually, use that one
+	if kafkaCluster.Spec.CruiseControlConfig.CapacityConfig != "" {
+		return kafkaCluster.Spec.CruiseControlConfig.CapacityConfig
+	}
+
+	capacityConfig := CruiseControlCapacityConfig{}
+
+	for _, brokerState := range kafkaCluster.Spec.Brokers {
+		if brokerState.BrokerConfig == nil {
+			continue
+		}
+		brokerCapacity := BrokerCapacity{
+			BrokerID: fmt.Sprintf("%d", brokerState.Id),
+			Capacity: Capacity{
+				DISK:  generateBrokerDisks(brokerState, kafkaCluster.Spec, log),
+				CPU:   storageConfigCPUDefaultValue,
+				NWIN:  storageConfigNWINDefaultValue,
+				NWOUT: storageConfigNWOUTDefaultValue,
+			},
+			Doc: "Capacity unit used for disk is in MB, cpu is in percentage, network throughput is in KB.",
+		}
+
+		capacityConfig.BrokerCapacities = append(capacityConfig.BrokerCapacities, brokerCapacity)
+	}
+
+	result, err := json.MarshalIndent(capacityConfig, "", "    ")
+	if err != nil {
+		log.Error(err, "Could not marshal cruise control capacity config")
+	}
+
+	return string(result)
+}
+
+func generateBrokerDisks(brokerState v1beta1.Broker, kafkaClusterSpec v1beta1.KafkaClusterSpec, log logr.Logger) map[string]string {
+	brokerDisks := map[string]string{}
+
+	// Get disks from the BrokerConfigGroup if it's in use
+	if brokerState.BrokerConfigGroup != "" {
+		brokerConfigGroup := kafkaClusterSpec.BrokerConfigGroups[brokerState.BrokerConfigGroup]
+		parseMountPathWithSize(brokerConfigGroup, log, brokerState, brokerDisks)
+	}
+
+	//Get disks from the BrokerConfig itself
+	parseMountPathWithSize(*brokerState.BrokerConfig, log, brokerState, brokerDisks)
+
+	return brokerDisks
+}
+
+func parseMountPathWithSize(brokerConfigGroup v1beta1.BrokerConfig, log logr.Logger, brokerState v1beta1.Broker, brokerDisks map[string]string) {
+	for _, storageConfig := range brokerConfigGroup.StorageConfigs {
+		int64Value, isConvertible := util.QuantityPointer(storageConfig.PvcSpec.Resources.Requests["storage"]).AsInt64()
+		if isConvertible == false {
+			log.Info("Could not convert 'storage' quantity to Int64 in brokerConfig for broker",
+				"brokerId", brokerState.Id)
+		}
+
+		brokerDisks[storageConfig.MountPath] = strconv.FormatInt(int64Value, 10)
```

Comment:
In https://github.com/banzaicloud/kafka-operator/blob/376d13ac9336c739cc793a3b9375e311070d6802/pkg/resources/kafka/configmap.go "/kafka" is appended to mountpath when building kafka log.dirs. CC is expecting exact path in capacity config iiuc. I think the suffix needs to be added here as well

Reply:
Thanks, good catch. In the future it would be great to move this into a single common place so we would avoid this bug in the future when we are working with mountpaths.